File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change to your UID

	// Callback function for date and time callback
	static void DateTimeCB(BrickletRealTimeClock sender, int year, byte month, byte day, byte hour,
	                       byte minute, byte second, byte centisecond, byte weekday, long timestamp)
	{
		Console.WriteLine("Year: " + year);
		Console.WriteLine("Month: " + month);
		Console.WriteLine("Day: " + day);
		Console.WriteLine("Hour: " + hour);
		Console.WriteLine("Minute: " + minute);
		Console.WriteLine("Second: " + second);
		Console.WriteLine("Centisecond: " + centisecond);
		Console.WriteLine("Weekday: " + weekday);
		Console.WriteLine("Timestamp: " + timestamp);
		Console.WriteLine("");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletRealTimeClock rtc = new BrickletRealTimeClock(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register date and time callback to function DateTimeCB
		rtc.DateTime += DateTimeCB;

		// Set period for date and time callback to 5s (5000ms)
		// Note: The date and time callback is only called every 5 seconds
		//       if the date and time has changed since the last call!
		rtc.SetDateTimeCallbackPeriod(5000);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/java/ExampleCallback.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletRealTimeClock;

public class ExampleCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Real-Time Clock Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletRealTimeClock rtc = new BrickletRealTimeClock(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add date and time listener
		rtc.addDateTimeListener(new BrickletRealTimeClock.DateTimeListener() {
			public void dateTime(int year, short month, short day, short hour,
			                     short minute, short second, short centisecond,
			                     short weekday, long timestamp) {
				System.out.println("Year: " + year);
				System.out.println("Month: " + month);
				System.out.println("Day: " + day);
				System.out.println("Hour: " + hour);
				System.out.println("Minute: " + minute);
				System.out.println("Second: " + second);
				System.out.println("Centisecond: " + centisecond);
				System.out.println("Weekday: " + weekday);
				System.out.println("Timestamp: " + timestamp);
				System.out.println("");
			}
		});

		// Set period for date and time callback to 5s (5000ms)
		// Note: The date and time callback is only called every 5 seconds
		//       if the date and time has changed since the last call!
		rtc.setDateTimeCallbackPeriod(5000);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
